bindings/go/oci/blob/artifact_blob.go

```diff
@@ -193,6 +195,60 @@ func (r *ArtifactBlob) OCIDescriptor() ociImageSpecV1.Descriptor {
 	}
 }
 
+// HasSizeAndDigest checks if the blob has both a pre-calculated size and a pre-calculated  digest.
+func (r *ArtifactBlob) HasSizeAndDigest() bool {
+	return r.HasPrecalculatedSize() && r.HasPrecalculatedDigest()
+}
+
+// NewArtifactWithBufferedBlob creates a new ArtifactBlob with an in-memory buffered blob.
+// It can be used to covert ArtifactBlob with unknown size or digest to a new instance where those fields a set.
+func (r *ArtifactBlob) NewArtifactWithBufferedBlob() (result *ArtifactBlob, err error) {
+	// Get current reader.
+	reader, err := r.ReadCloser()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get blob reader: %w", err)
+	}
+	defer func() {
+		err = errors.Join(err, reader.Close())
+	}()
+
+	// Get current media type, even if it is empty.
+	mediaType, _ := r.MediaType()
+
+	// Create buffered in-memory blob and load it.
+	inMemoryBlob := inmemory.New(reader, inmemory.WithMediaType(mediaType))
+	err = inMemoryBlob.Load()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load in-memory blob: %w", err)
+	}
+
+	// Create a new instance of the descriptor.
+	var newArtifact descriptor.Artifact
```

Comment:
I really don't like this, can we make this method available on Artifact instead?